### PR TITLE
Update eslint, babel, jest, underreact, add browserslist

### PR DESCRIPTION
**Pull request by katydecorah**
This PR updates a few devDependencies and adds browserslist to package.json:

- Update eslint
- Update babel
- Update jest (to support babel update)
- Updated underreact (to support babel update)
- Add browserslist 
- Update node version in .travis.yml (to support eslint update)

By adding browerslist, the package reduced in size!

Current | Proposed
---|---
npm notice package size:  70.1 kB<br/>npm notice unpacked size: 459.9 kB  |npm notice package size:  60.3 kB<br/>npm notice unpacked size: 302.8 kB

### CODE REVIEW

File: src/docs/public/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
```
Comment (katydecorah) [inline src/docs/public/.gitignore:1]: Due to this bug https://github.com/mapbox/underreact/issues/100, this addition attempts to keep the `public` folder in place.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,2 @@
 language: node_js
-node_js:
-  - 8
-before_install:
-  - npm install npm@6 -g
-install:
-  - npm ci
+node_js: 12
```
Comment (katydecorah) [inline .travis.yml:2]: Should we go with 14?

File: src/docs/babel.config.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const config = require('../../babel.config.js');
```
Comment (katydecorah) [inline src/docs/babel.config.js:1]: Underreact threw a lot of babel-related warnings, so this PR adds this config closer to the underreact config which soothed the warnings.

File: src/components/control-card/__tests__/__snapshots__/control-card.test.js.snap [hunk 4]
```diff
@@ -46,7 +46,7 @@ exports[`ControlCard collapsible onButtonClick is called 1`] = `
             ],
             "results": Array [
               Object {
-                "isThrow": false,
+                "type": "return",
```
Comment (katydecorah) [inline src/components/control-card/__tests__/__snapshots__/control-card.test.js.snap:49]: I'm not confident if this change is ok, but the component appears to work as expected.

File: .travis.yml (hunk 2 above)
Comment (samanpwbb) [reply .travis.yml:2]: Ya, will save us another upgrade soon

**Review by samanpwbb — APPROVED** [review]
Ran the tests and build scripts and everything works great – thanks for getting a browserlist definition in here!

**Pull request merged**